probes/blocksDeleteOnBranches/impl_test.go

```diff
@@ -0,0 +1,185 @@
+// limitations under the License.
+
+//nolint:stylecheck
+package blocksDeleteOnBranches
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	"github.com/ossf/scorecard/v4/checker"
+	"github.com/ossf/scorecard/v4/clients"
+	"github.com/ossf/scorecard/v4/finding"
+)
+
+func Test_Run(t *testing.T) {
+	t.Parallel()
+	trueVal := true
+	falseVal := false
+	branchVal1 := "branch-name1"
+	branchVal2 := "branch-name1"
+
+	//nolint:govet
+	tests := []struct {
+		name     string
+		raw      *checker.RawResults
+		outcomes []finding.Outcome
+		err      error
+	}{
+		{
+			name: "Blocks deletions on 1/1 branches",
+			raw: &checker.RawResults{
+				BranchProtectionResults: checker.BranchProtectionsData{
+					Branches: []clients.BranchRef{
+						{
+							Name: &branchVal1,
+							BranchProtectionRule: clients.BranchProtectionRule{
+								AllowDeletions: &falseVal,
+							},
+						},
+					},
+				},
+			},
+			outcomes: []finding.Outcome{
+				finding.OutcomePositive,
+			},
+		},
+		{
+			name: "Blocks deletions on 2/2 branches",
+			raw: &checker.RawResults{
+				BranchProtectionResults: checker.BranchProtectionsData{
+					Branches: []clients.BranchRef{
+						{
+							Name: &branchVal1,
+							BranchProtectionRule: clients.BranchProtectionRule{
+								AllowDeletions: &falseVal,
+							},
+						},
+						{
+							Name: &branchVal2,
+							BranchProtectionRule: clients.BranchProtectionRule{
+								AllowDeletions: &falseVal,
+							},
+						},
+					},
+				},
+			},
+			outcomes: []finding.Outcome{
+				finding.OutcomePositive, finding.OutcomePositive,
+			},
+		},
+		{
+			name: "Blocks deletions on 1/2 branches - 1",
+			raw: &checker.RawResults{
+				BranchProtectionResults: checker.BranchProtectionsData{
+					Branches: []clients.BranchRef{
+						{
+							Name: &branchVal1,
+							BranchProtectionRule: clients.BranchProtectionRule{
+								AllowDeletions: &falseVal,
+							},
+						},
+						{
+							Name: &branchVal2,
+							BranchProtectionRule: clients.BranchProtectionRule{
+								AllowDeletions: &trueVal,
+							},
+						},
+					},
+				},
+			},
+			outcomes: []finding.Outcome{
+				finding.OutcomePositive, finding.OutcomeNegative,
+			},
+		},
+		{
+			name: "Blocks deletions on 1/2 branches - 2",
+			raw: &checker.RawResults{
+				BranchProtectionResults: checker.BranchProtectionsData{
+					Branches: []clients.BranchRef{
+						{
+							Name: &branchVal1,
+							BranchProtectionRule: clients.BranchProtectionRule{
+								AllowDeletions: &trueVal,
+							},
+						},
+						{
+							Name: &branchVal2,
+							BranchProtectionRule: clients.BranchProtectionRule{
+								AllowDeletions: &falseVal,
+							},
+						},
+					},
+				},
+			},
+			outcomes: []finding.Outcome{
+				finding.OutcomeNegative, finding.OutcomePositive,
+			},
+		},
+		{
+			name: "Blocks deletions on 0/2 branches",
+			raw: &checker.RawResults{
```

Comment:
I think this test name is misleading. We don't have data available for one of them. I'd recommend renaming it "data not available" or something like that, and just look for the one OutcomeNotAvailable

Reply:
https://github.com/ossf/scorecard/pull/3691/commits/d7c16773ae85b5c49cffe37a49f8330e60bdd631